### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by donizer**
[DEMO LINK](https://donizer.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,263 @@
...
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {!!todos.length && (
+            <button
+              type="button"
+              className={classnames('todoapp__toggle-all', {
+                active: !todos.some((todo) => todo.completed === false),
+              })}
+              data-cy="ToggleAllButton"
+              onClick={() => toggleAll()}
```
Comment (lerastarynets) [inline src/App.tsx:162]:
```suggestion
              onClick={toggleAll}
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,263 @@
...
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {!!todos.length && (
+            <button
+              type="button"
+              className={classnames('todoapp__toggle-all', {
+                active: !todos.some((todo) => todo.completed === false),
```
Comment (lerastarynets) [inline src/App.tsx:159]:
```suggestion
                active: !todos.some((todo) => !todo.completed),
```

File: src/Contexts/AppContextProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      setVisibleTodos(response);
+    } catch {
+      setErrorMessage('Unable to load todos');
+    }
+  }, [setErrorMessage, setTodos, setVisibleTodos]);
+
+  useEffect(() => {
+    initialLoad();
+    inputRef.current?.focus();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (lerastarynets) [inline src/Contexts/AppContextProvider.tsx:72]: this rule shouldn't be disabled

**Review by lerastarynets — CHANGES_REQUESTED** [review]
I see you did a hard work) Let's improve it a little bit

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,265 @@
...
+                  data-cy="FilterLinkActive"
+                  onClick={() => setQueryCondition('active')}
+                >
+                  Active
+                </a>
+
+                <a
+                  href="#/completed"
+                  className={classnames('filter__link', {
+                    selected: queryCondition === 'completed',
```
Comment (sTorba24) [inline src/App.tsx:222]: create an enum for filter types

**Comment by donizer** [comment]
Pre-last commit failed only one test, looks like latency issue. So i modified time to disappear for error a bit.

**Pull request closed without merging**